client/src/views/Recordings.vue

```diff
@@ -58,79 +74,133 @@ export default defineComponent({
         },
         {
           title: 'Details',
-          key:'comments'
+          key: 'comments'
         },
-
         {
-            title:'Annotated by Me',
-            key:'userMadeAnnotations',
+          title: 'Annotated by Me',
+          key: 'userMadeAnnotations',
         },
     ]);
+    const dataLoading = ref(false);
     const fetchRecordings = async () => {
-        const recordings = await getRecordings();
-        recordingList.value = recordings.data;
-        // If we have a spectrogram being generated we need to refresh on an interval
-        let missingSpectro = false;
-        for (let i =0; i< recordingList.value.length; i+=1) {
-          if (!recordingList.value[i].hasSpectrogram) {
-            missingSpectro = true;
-            break;
-          }
+      dataLoading.value = true;
+      const recordings = await getRecordings();
+      recordingList.value = recordings.data;
+      // If we have a spectrogram being generated we need to refresh on an interval
+      let missingSpectro = false;
+      for (let i =0; i< recordingList.value.length; i+=1) {
+        if (!recordingList.value[i].hasSpectrogram) {
+          missingSpectro = true;
+          break;
+        }
+      }
+      if (missingSpectro) {
+        if (intervalRef === null) {
+          intervalRef = setInterval(() => fetchRecordings(), 5000);
         }
-        if (missingSpectro) {
-          if (intervalRef === null) {
-            intervalRef = setInterval(() => fetchRecordings(), 5000);
-          }
-        } else  {
-          if (intervalRef !== null) {
-            clearInterval(intervalRef);
-          }
+      } else  {
+        if (intervalRef !== null) {
+          clearInterval(intervalRef);
         }
-        const shared = await getRecordings(true);
-        sharedList.value = shared.data;
+      }
+      const shared = await getRecordings(true);
+      sharedList.value = shared.data;
+      dataLoading.value = false;
+    };
 
+    const fetchRecordingTags = async () => {
+      dataLoading.value = true;
+      const tags = await getRecordingTags();
+      recordingTagList.value = tags.data;
+      dataLoading.value = false;
     };
-    onMounted(() => fetchRecordings());
+
+    const filterTags: Ref<string[]> = ref([]);
+    const sharedFilterTags: Ref<string[]> = ref([]);
+    const recordingTags = computed(() => {
+      const tags = recordingList.value
+        .map((recording: Recording) => recording.tag_text)
+        .filter((tag: string | null) => tag !== null);
+      return [...new Set(tags)];
+    });
+    const sharedRecordingTags = computed(() => {
+      const tags = sharedList.value
+        .map((recording: Recording) => recording.tag_text)
+        .filter((tag: string | null) => tag !== null);
+      return [...new Set(tags)];
+    });
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const tagFilter: FilterFunction = (value: string, search: string, item: any) => {
```

Comment:
I know that the Vuetify demo has a `any` tag for the type but there is an actual type specification for customFilters.

```suggestion
    const tagFilter: FilterFunction = (value: string, search: string, item?: InternalItem<Recording>) => {
```

This would also require further down swapping from `item?.columns?.tag_text` to `item?.raw.tag_text`.  I don't know what `columns` is and why it is included in the item but it isn't referenced in type specification in Vuetify so I don't know if it will always be supported and may be removed in the future.

I think probably the same thing should be done in the `sharedTagFilter` as well.